src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -861,8 +873,14 @@ def find_through_authorization_code_flow(self, tenant, resource, authority_url):
 
         # exchange the code for the token
         context = self._create_auth_context(tenant)
-        token_entry = context.acquire_token_with_authorization_code(results['code'], results['reply_url'],
-                                                                    resource, _CLIENT_ID, None)
+        try:
+            token_entry = context.acquire_token_with_authorization_code(results['code'], results['reply_url'],
+                                                                        resource, _CLIENT_ID, None)
+        except InvalidURL as err:
+            logger.debug('Invalid url when acquiring token with authorization code:%s\n%s',
+                         results, traceback.format_exc())
+            raise UnknownError('Invalid url when acquiring token with authorization code: %s' %
+                               err.request if hasattr(err, 'request') else '')
```

Comment:
Consider unifying the code in one place to avoid duplication.

Reply:
As the `acquire_token` fuction diffs for each scenario, the debug log and error msg are also customized. So I didn't unify the code.